content/language/R/deploy.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Test your R deployment
+keywords: deploy, kubernetes, R
+description: Learn how to develop locally using Kubernetes
+---
+
+## Prerequisites
+
+- Complete all the previous sections of this guide, starting with [Containerize a R application](containerize.md).
+- [Turn on Kubernetes](/desktop/kubernetes/#install-and-turn-on-kubernetes) in Docker Desktop.
+
+## Overview
+
+In this section, you'll learn how to use Docker Desktop to deploy your application to a fully-featured Kubernetes environment on your development machine. This allows you to test and debug your workloads on Kubernetes locally before deploying.
+
+## Create a Kubernetes YAML file
+
+In your `r-docker-dev` directory, create a file named
+`docker-r-kubernetes.yaml`. Open the file in an IDE or text editor and add
+the following contents. Replace `DOCKER_USERNAME/REPO_NAME` with your Docker
+username and the name of the repository that you created in [Configure CI/CD for
+your R application](configure-ci-cd.md).
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: docker-r-demo
+  namespace: default
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      service: shiny
+  template:
+    metadata:
+      labels:
+        service: shiny
+    spec:
+      containers:
+       - name: shimy-service
+         image: DOCKER_USERNAME/REPO_NAME
+         imagePullPolicy: Always
+         env:
+          - name: POSTGRES_PASSWORD
+            value: mysecretpassword
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: service-entrypoint
+  namespace: default
+spec:
+  type: NodePort
+  selector:
+    service: shiny
+  ports:
+  - port: 3838
+    targetPort: 3838
+    nodePort: 30001
+```
+
+In this Kubernetes YAML file, there are two objects, separated by the `---`:
+
+ - A Deployment, describing a scalable group of identical pods. In this case,
+   you'll get just one replica, or copy of your pod. That pod, which is
+   described under `template`, has just one container in it. The
+    container is created from the image built by GitHub Actions in [Configure CI/CD for
+    your R application](configure-ci-cd.md).
+ - A NodePort service, which will route traffic from port 30001 on your host to
+   port 3838 inside the pods it routes to, allowing you to reach your app
+   from the network.
+
+To learn more about Kubernetes objects, see the [Kubernetes documentation](https://kubernetes.io/docs/home/).
+
+## Deploy and check your application
+
+1. In a terminal, navigate to `r-docker-dev` and deploy your application to
+   Kubernetes.
+
+   ```console
+   $ kubectl apply -f docker-r-kubernetes.yaml
+   ```
+
+   You should see output that looks like the following, indicating your Kubernetes objects were created successfully.
+
+   ```shell
```

Comment:
this should probably be `text`, not `shell` (which is equivalent to `bash`)

Reply:
Changed in `text`